backend/Testing/Browser/EmailWorkflowTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Testing.Browser.Base;
+using Testing.Browser.Page;
+using Testing.Browser.Page.External;
+using Testing.Browser.Util;
+
+namespace Testing.Browser;
+
+[Trait("Category", "Integration")]
+public class EmailWorkflowTests : PageTest
+{
+    [Fact]
+    public async Task RegisterVerifyUpdateVerifyEmailAdress()
+    {
+        // Step: Register
+        var mailinatorId = Guid.NewGuid().ToString();
+        var email = $"{mailinatorId}@mailinator.com";
+        var password = email;
+        await using var userDashboardPage = await RegisterUser($"Test: Verify email - {mailinatorId}", email, password);
+
+        await userDashboardPage.EmailVerificationAlert.AssertPleaseVerify();
+
+        // Step: Request extra verification email and verify
+        await userDashboardPage.EmailVerificationAlert.ClickResendEmail();
+        await userDashboardPage.EmailVerificationAlert.AssertVerificationSent();
+
+        // await Page.PauseAsync(); // Pause in dev to forward e-mails to mailinator
+
+        MailInboxPage inboxPage = IsDev
```

Comment:
let's make a factory function out of this, then we can use it here and down below. You can declare static methods with an implementation on the `MailInboxPage` interface.